Add a tool to check for missing docs redirects

ptgott opened the pull request:
Add a tool to check for URL paths in the Teleport docs site that do not correspond to actual docs pages or redirects. These URLs are most likely links that will 404 if a user follows them. This way, we can be more strategic about the redirects that we configure for the docs site, meaning that we do not need to add redirects for every single docs page that we rename or delete. A later change can add a GitHub Actions workflow to run the program.

ptgott committed:
Add a tool to check for missing docs redirects

Add a tool to check for URL paths in the Teleport docs site that do not
correspond to actual docs pages or redirects. These URLs are most likely
links that will 404 if a user follows them. This way, we can be more
strategic about the redirects that we configure for the docs site,
meaning that we do not need to add redirects for every single docs page
that we rename or delete. A later change can add a GitHub Actions
workflow to run the program.

avatus commented on docs/check-redirects/check-redirects.js:
```diff
@@ -0,0 +1,120 @@
+    );
+    const text = this.fs.readFileSync(filePath, 'utf8');
+    const docsURLs = [...text.matchAll(docsPattern)];
+    if (!docsURLs) {
+      return;
+    }
```
should explicitly check for length here (docsURLs is an array yes?)

avatus commented on docs/check-redirects/check-redirects.js:
```diff
@@ -0,0 +1,120 @@
+    let result = [];
+    docsURLs.forEach(url => {
+      const docsPath = this.urlToDocsPath(url[0]);
+      const entry = this.fs.statSync(docsPath, {
+        throwIfNoEntry: false,
+      });
```
this seems like it should be named `missingEntry` ? if the entry is undefined, we do something right? or maybe like, `entryExists` idk. maybe just a comment "if entry exists, we skip"

ptgott commented:
TODO: I also need to clarify in the README what the `--in` flag should point to.

ptgott committed:
Respond to avatus feedback

- Clarify the `--in` flag.
- Add a more descriptive name for missing directory entries.
- Explicitly check `docsURLs` length
- Use a Set for the internal redirects field

avatus reviewed (approved):
Looks great! Thanks
